### DISCUSSION TITLE
Older world loading/converting to 1.18 World in Minestrom

### DISCUSSION THREAD

**Discussion by JackM33** [Minestom/Minestom#837] [Help]
Hi I'm new to the Minestrom api so forgive me if all of this should be obvious but I've currently set up a dirt simple server/instance that loads and saves chunks to a given world to which I've copied and pasted region .mca files over from an existing world I've exported from World Painter (which is yet to fully update to support 1.18) so it's effectively a 1.15 world. When I load the world in minecraft single player it converts everything just fine to 1.18 but when I load the world in Minestrom certain blocks are missing for instance jungle leaves on my trees as show in the screenshot. 1.18 blocks do seem to load though so it seems to be an issue converting from the older world to the newer version. Is this simply not part of Minestrom yet or should I be telling it to interpret the world differently at first?

If I first load the chunks in single player and then copy those files to the Minestromworld folder, those chunks load perfectly fine while the rest of the world is missing things. A workaround would be to load every piece of the world in single player first and then copy that over to Minestrom but if possible I'd hope to avoid that hassle.

I was looking in to Vanillink as a possible solution. 

![2022-03-30_14 19 44](https://user-images.githubusercontent.com/47346677/160933807-7870b7c2-a6a1-4a48-a4b0-ab853307c572.png)

![2022-03-30_14 19 59](https://user-images.githubusercontent.com/47346677/160933836-0c242859-907a-4126-bc37-31cecaa8e720.png)

This is a screenshot of some chunks I had first loaded in single player before copying them over to Minestrom to show that those chunks do correctly load the leaves whereas the ones next to them that I did not load fail to load the leaves.

Thank you very much! I really appreciate any help!

**Comment by cody-quinn** [comment]
Minestom doesn't handle world version conversions, I would first open the world in vanilla single player then use the resulting world in your .minecraft for Minestom.

**Reply by LooFifteen** [reply]
^^ You can use 1.16+'s Optimize World feature. Or just load all the chunks..

**Reply by JackM33** [reply]
Ok got it thank you very much! Yeah I actually had no idea that feature existed in the edit menu. Unfortunately when I run optimize it runs fine but then upon loading with Minestrom it effectively becomes a superflat world as far as I can tell. I'll probably just resort to loading all the chunks in single player first but on any off chance do you have any clue why that might occur?

Thank you both! I very much appreciate it.

**Reply by RinesThaix** [reply]
When I used world optimization feature with my world before loading into Minestom, it still had blocks, which are supposed to have nbt data, broken (like fences were not connected). So what I did is determined exact chunk range I would like to load, wrote a spigot plugin which iterates through that range and loads + unloads + saves every single chunk. After that, world can be imported without any issues.

**Comment by jglrxavpok** [comment]
Are you able to send the world in this discussion so we can investigate further?

**Reply by JackM33** [reply]
Hi unfortunately the world is supposed to remain private but I may be able to create a simple example world that demonstrates the same issue if you think there's any reason to believe it could be solved without simply loading all the chunks as mentioned by the other answers?

Thanks
